woocommerce/wc-admin
# Home Screen - modified tasks extensibility

Fixes #5242

This PR adds the modifications required for the task list extensibility.
 
* It should allow 3pds to create setup tasks for their extensions with the following items: task name (with link to an extension page), time to complete the tasks, and a dismiss feature
* If a merchant completes the task list and then installs an extension that has a setup task associated, then the task list will be displayed again with that extension.

### Accessibility




### Screenshots
![screenshot-one wordpress test-2020 12 02-17_01_48](https://user-images.githubusercontent.com/1314156/100925052-3c99a700-34c0-11eb-9a32-f0eda5c7c75a.png)


### Detailed test instructions:
1. Run `npm run example -- --ext=add-task`.
2. Activate the plugin `WooCommerce Admin Add Task Example`.
3. Be sure the setup task list is not hidden
```
UPDATE `wp_options` SET `option_value` = 'no' WHERE `wp_options`.`option_name` = 'woocommerce_task_list_hidden';
```
4. Go to the `Home` screen.
5. Verify the task list `Extensions setup` is visible.
6. Try the `Dismiss` button and undo the dismission. Verify everything is working as expected.

7. Hide the setup list (`Finish store setup`) and verify the other list is still there.
8. Hide the `Extensions setup` list and verify that it's not visible anymore. 
![screenshot-one wordpress test-2020 12 02-18_49_37](https://user-images.githubusercontent.com/1314156/100935763-470f6d00-34cf-11eb-98a2-b83fc0dac808.png)
9. Try adding another task * and verify the list now is visible again.
10. Confirm the callback method `onDismiss` is working correctly when a task is dismissed. To do it just take a look at the browser's console after dismissing a task, the message: `The task 'example-[number]' was dismissed
` should be visible.

_* I did that in the branch `add/5242_test` so you can check out that branch and run `npm run example -- --ext=add-task-two`, and activate the plugin `WooCommerce Admin Add Task Example Two`. The task button `Mark as complete` should mark all these tasks as completed after being pressed._
